src/main/java/io/leonis/subra/math/CombineByPlayerStrategySupplier.java

```diff
@@ -0,0 +1,42 @@
+package io.leonis.subra.math;
+
+import io.leonis.subra.game.data.*;
+import java.util.Map.Entry;
+import java.util.function.BinaryOperator;
+import java.util.stream.Collectors;
+import lombok.experimental.Delegate;
+
+/**
+ * The Class CombineByPlayerStrategySupplier.
+ *
+ * This class represents the computation of combining two {@link Strategy.Supplier} by
+ * {@link io.leonis.subra.game.data.Player.PlayerIdentity}.
+ *
+ * @author Rimon Oz
+ */
+public class CombineByPlayerStrategySupplier implements Strategy.Supplier {
+  @Delegate
+  private final Strategy.Supplier strategy;
+
+  /**
+   * Computes the {@link Strategy.Supplier} as a result of combining the supplied strategies by
+   * {@link io.leonis.subra.game.data.Player.PlayerIdentity}.
+   *
+   * @param leftMap  The first strategy to combine.
+   * @param rightMap The second strategy to combine.
+   * @param combiner The {@link BinaryOperator combinator} function.
+   */
+  public CombineByPlayerStrategySupplier(
```

Comment:
What about `GroupByPlayerStrategySupplier` ?

Reply:
because the result is a `Map<PlayerIdentity, PlayerCommand>`, not a `Map<PlayerIdentity, Collection<PlayerCommand>>`

Reply:
It might be if you make the return type of the combiner a generic :)

Reply:
but then it's no longer a `Strategy.Supplier`

Reply:
I missed that. good catch.. I guess it's going to be called `CombineByPlayerStrategySupplier` then :P until we find a way to implement Strategy without being a supplier.